response_operations_ui/templates/account/activate-account.html

```diff
@@ -29,18 +29,30 @@
                 {% endcall %}
             {% endif %}
         {% endwith %}
-          {% if form.errors %}
+          {% set password_error_count = 0 %}
+          {% set password_confirm_error_count = 0 %}
+          {% if form.password.errors %}
+            {% set unique_errors = form.password.errors | unique %}
+            <p>{{unique_errors}}</p>
+            {% set password_error_count = form.password.errors | length|default(0, true) %}
+          {% endif %}
+          {% if form.password_confirm.errors %}
+            {% set password_confirm_error_count = form.password_confirm.errors | length|default(0, true) %}
+          {% endif %}
+          {% if password_error_count or password_confirm_error_count %}
             {% set errorData = [] %}
-                {% set error_count = form.errors | length %}
+            {% set password = namespace(text = '') %}
+            {% set password_confirm = namespace(text = '') %}
+                {% set error_count = password_error_count + password_confirm_error_count %}
                 {% set plural_suffix = 's' if error_count > 1 else '' %}
-                {% set header_text = 'This page has ' ~ error_count ~ ' error' ~ plural_suffix %}
+                {% set plural_singular_word = 'are ' if error_count > 1 else 'is ' %}
+                {% set header_text = 'There ' ~ plural_singular_word ~ error_count ~ ' problem' ~ plural_suffix ~ ' with your password' %}
```

Comment:
This seems overly complicated for just a count, there seems to be 11 sets made, it is also repeated elsewhere, can it not be simplified and refactored into one?  I will have a play and see if I can recommend anything

Reply:
You are right. I have simplified this.